node/node.go

```diff
@@ -611,7 +620,11 @@ func (n *Node) ConfigureRPC() {
 	rpccore.SetEvidencePool(n.evidencePool)
 	rpccore.SetP2PPeers(n.sw)
 	rpccore.SetP2PTransport(n)
-	rpccore.SetPubKey(n.privValidator.GetPubKey())
+	pubKey, err := n.privValidator.GetPubKey()
+	if err != nil {
+		n.Logger.Error("Error configuring RPC. Can not retrieve privValidator's public key", "err", err)
```

Comment:
same here. too many different variations of a single thing

Reply:
that sounds even better!

Reply:
Done in https://github.com/tendermint/tendermint/pull/2948/commits/5c09275691b2190d8747f4f1b1817f63114b1c6b
Besides the wrapping in remote signer only the logger messages are left to give more context.